src/libraries/Microsoft.PowerFx.Core/Lexer/TexlLexer.cs

```diff
@@ -394,6 +399,32 @@ public List<Token> GetTokens(string text, Flags flags = Flags.None)
             }
 
             return tokens;
+        }
+
+        /// <summary>
+        /// In case of empty string scritp and TextFirst is active, returns a list of tokens representing an empty string.
+        /// </summary>
+        /// <param name="text">Script.</param>
+        /// <param name="flags">Parser flags.</param>
+        /// <param name="tokens">Tokens object initialized by the caller.</param>
+        /// <returns></returns>
+        public bool TryParseTextFirstEmptyString(string text, Flags flags, out List<Token> tokens)
+        {
+            if (text.Length == 0 && flags.HasFlag(TexlLexer.Flags.TextFirst))
+            {
+                var span = new Span(0, 0);
+                tokens = new List<Token>
+                {
+                    new StrInterpStartToken(span, true),
+                    new StrInterpEndToken(span, true),
+                    new EofToken(span)
+                };
+
+                return true;
+            }
+
+            tokens = new List<Token>();

```

Comment:
>           tokens = new List<Token>(); [](http://example.com/codeflow?start=1&length=38)

nit - avoid this. 
We're relying on side-effects of a function/ 
If we rreturn false, then assume out params are all null